sympy/core/tests/test_sympify.py

```diff
@@ -512,6 +513,51 @@ def test_sympify_set():
     assert sympify(set()) == EmptySet()
 
 
+def test_numpy():
+    from sympy.utilities.pytest import skip
+    np = import_module('numpy')
+
+
+    def equal(x, y):
+        return x == y and type(x) == type(y)
+
+    if not np:
+        skip('numpy not installed.Abort numpy tests.')
+
+    assert sympify(np.bool_(1)) is S(True)
+    assert equal(
+        sympify(np.int_(1234567891234567891)), S(1234567891234567891))
+    assert equal(sympify(np.intc(1234567891)), S(1234567891))
+    assert equal(
+        sympify(np.intp(1234567891234567891)), S(1234567891234567891))
+    assert equal(sympify(np.int8(-123)), S(-123))
+    assert equal(sympify(np.int16(-12345)), S(-12345))
+    assert equal(sympify(np.int32(-1234567891)), S(-1234567891))
+    assert equal(
+        sympify(np.int64(-1234567891234567891)), S(-1234567891234567891))
+    assert equal(sympify(np.uint8(123)), S(123))
+    assert equal(sympify(np.uint16(12345)), S(12345))
+    assert equal(sympify(np.uint32(1234567891)), S(1234567891))
+    assert equal(
+        sympify(np.uint64(1234567891234567891)), S(1234567891234567891))
+    assert equal(sympify(np.float16(1.123)), S(1.123))
+    assert equal(sympify(np.float32(1.1234567)), S(1.1234567))
+    assert equal(sympify(np.float64(1.1234567891234)), S(1.1234567891234))
```

Comment:
I'm trying to use `numpy.finfo(numpy.float).precision` to supply required no. of max digits to the Float constructor. However, it seems to be inadequate. For example `numpy.float16` can handle max no. of 4 decimal digits, but precision as reported by `numpy.finfo` is 3. It's also 1 or 2 units lower for the other float types as well.